capability-providers/keyvalue/rust/src/main.rs

```diff
@@ -0,0 +1,168 @@
+//! # wasmCloud Redis Provider
+//!
+//! Topics relevant to a capability provider:
+//!
+//! RPC:
+//!   * wasmbus.rpc.{prefix}.{provider_key}.{link_name} - Get Invocation, answer InvocationResponse
+//!   * wasmbus.rpc.{prefix}.{public_key}.{link_name}.linkdefs.get - Query all link defs for this provider. (queue subscribed)
+//!   * wasmbus.rpc.{prefix}.{public_key}.{link_name}.linkdefs.del - Remove a link def. Provider de-provisions resources for the given actor.
+//!   * wasmbus.rpc.{prefix}.{public_key}.{link_name}.linkdefs.put - Puts a link def. Provider provisions resources for the given actor.
+//!   * wasmbus.rpc.{prefix}.{public_key}.{link_name}.shutdown - Request for graceful shutdown
+
+#[macro_use]
+extern crate log;
+#[macro_use]
+extern crate lazy_static;
+
+lazy_static! {
+    static ref LINKDEFS: RwLock<HashMap<String, LinkDefinition>> = RwLock::new(HashMap::new());
```

Comment:
Ha, look at how much work you need to make this accessible across the file! You should've written this in Go instead where you can do anything 🚗